panel/main.cpp

```diff
@@ -48,65 +49,32 @@ void termSignalHandler(int)
         qApp->quit();
 }
 
-
-void printHelp()
-{
-    QTextStream out(stdout);
-    out << "Usage: lxqt-panel [options]" << endl;
-    out << endl;
-    out << "Start lxde-qt panel and its plugins" << endl;
-    out << endl;
-    out << "Options:" << endl;
-    out << "  -h, --help                    Show help about options" << endl;
-    out << "      --version                 Show version information" << endl;
-    out << "  -c, --configfile=CONFIGFILE   Use alternate configuration file" << endl;
-}
-
-
-void printVersion()
-{
-    QTextStream out(stdout);
-    out << "lxqt-panel " << LXQT_VERSION << endl;
-}
-
-
 int main(int argc, char *argv[])
 {
-    QString configFile;
-    for (int i=1; i < argc; ++i)
-    {
-        QString arg = QString::fromLocal8Bit(argv[i]);
 
-        if (arg == "--help" || arg == "-h")
-        {
-            printHelp();
-            return 0;
-        }
+    LxQtPanelApplication app(argc, argv);
+    QApplication::setApplicationName("lxqt-panel");
+    QApplication::setApplicationVersion(LXQT_VERSION);
 
-        if (arg == "--version")
-        {
-            printVersion();
-            return 0;
-        }
+    QCommandLineParser parser;
+    parser.setApplicationDescription("lxde-qt panel");
+    parser.addHelpOption();
+    parser.addVersionOption();
 
-        if (arg == "-c" || arg.startsWith("--conf"))
-        {
-            if (i+1 < argc)
-            {
-                configFile = QString::fromLocal8Bit(argv[i+1]);
-            }
-        }
-    }
+    QCommandLineOption configFileOption(QStringList()
+            << QStringLiteral("c") << QStringLiteral("config") << QStringLiteral("configfile"),
+            QCoreApplication::translate("main", "Use alternate configuration file."),
+            QCoreApplication::translate("main", "Configuration file"));
+    parser.addOption(configFileOption);
 
-    QScopedPointer<LxQtPanelApplication> app(new LxQtPanelApplication(argc, argv, configFile));
+    parser.process(app);
 
+    app.init(parser.value(configFileOption));
```

Comment:
Please, do not add an init method. We should force passing the config file to the constructor. For `parser.process()`, you can pass the argument list instead of the QApplication itself. You can use `QCoreApplication::arguments()`.